misc/config_definitions.json

```diff
@@ -3502,6 +3502,12 @@
                 "custom8": "Custom 8"
             }
         },
+        "http_auth_header": {
+            "group": "auth",
+            "section": "general",
+            "order": 4,
+            "type": "text"
```

Comment:
Can you set the default to REMOTE_USER and streamline getExternalUsername?
We also need to add the text for this option in `resources/lang/en/settings.php`